File: GreedyStarAssistant/app/src/main/java/com/greedy/assistant/TapService.java
package com.greedy.assistant;

import android.accessibilityservice.AccessibilityService;
import android.graphics.Path;
import android.os.Build;
import android.view.accessibility.AccessibilityEvent;

public class TapService extends AccessibilityService {
    public static TapService instance = null;

    public static void tap(int x, int y) {
        if (instance != null) instance.performGlobalClick(x, y);
    }

    @Override
    public void onServiceConnected() {
        instance = this;
    }

    private void performGlobalClick(int x, int y) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            Path path = new Path();
            path.moveTo(x, y);
            GestureDescription.Builder builder = new GestureDescription.Builder();
            builder.addStroke(new GestureDescription.StrokeDescription(path, 0, 100));
            dispatchGesture(builder.build(), null, null);
        }
    }

    @Override
    public void onAccessibilityEvent(AccessibilityEvent event) {}
    @Override
    public void onInterrupt() {}
}

File: GreedyStarAssistant/app/src/main/java/com/greedy/assistant/TapService.kt
package com.greedy.assistant

import android.accessibilityservice.AccessibilityService
import android.graphics.Path
import android.os.Build
import android.view.accessibility.AccessibilityEvent
import kotlinx.coroutines.*

class TapService : AccessibilityService() {
    companion object {
        var instance: TapService? = null
        fun tap(x: Int, y: Int, duration: Long = 100) {
            instance?.performGlobalClick(x, y, duration)
        }
    }

    override fun onServiceConnected() {
        instance = this
    }

    private fun performGlobalClick(x: Int, y: Int, duration: Long) {
        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
            val path = Path().apply { moveTo(x.toFloat(), y.toFloat()) }
            val gesture = GestureDescription.Builder()
                .addStroke(GestureDescription.StrokeDescription(path, 0, duration))
                .build()
            dispatchGesture(gesture, null, null)
        }
    }

    override fun onAccessibilityEvent(event: AccessibilityEvent?) {}
    override fun onInterrupt() {}
}